Update release notes for start of Aug

thebenrob opened the pull request:
#### Purpose
Bring release notes up to date for the beginning of August.

#### Changes
Added release notes for 5/21, 5/29, and 7/30

Commit:
Update release notes for start of Aug

perlmonger42 commented on api/release_notes/2019-release-notes.md:
```diff
@@ -31,10 +57,10 @@ rules/:rule_id?include=rule_components
 ### 3) Rule Components belong to Properties now
 
-Rule Components used to exist as children of a Rule.  That made it impossible to reuse the same components across multiple rules.  In order to enable us to add this capability in the future, Rule Components need to move out from under a Rule and live at the Property level.
+Rule Components used to exist as a child of a Rule.  That made it impossible to reuse the same components across multiple rules.  In order to enable us to add this capability in the future, Rule Components need to move out from under a Rule and live at the Property level.
```
Did you mean to change this back? or maybe you're just accidentally overwriting the change?

spencerroan replied:
I think `children` is grammatically correct.

thebenrob replied:
Seeing some weirdness, I'm sure it's user error.  Changed to children.

perlmonger42 reviewed (changes requested):
`Rule Components used to exist as a child of a Rule.` mixes a plural `Rule Components` with singular `child`.

Otherwise, looks good.

spencerroan commented on api/release_notes/2019-release-notes.md:
```diff
@@ -4,6 +4,32 @@ title: 2019 Release Notes
+If you are passing a `meta` block on a PATCH request, any `attributes` that you pass on the same request are not required and will be ignored.  This change will prevent modifying a resource and transitioning it to a new state at the same time.  This will allow you to get a resource back from the API, and transition its state without having to explicitly remove the `attributes` block before you PA ...
+
+* Libraries - cannot be modified and transitioned to a new state at the same time
+* Extensions, Rules, and Data Elements - cannot be modified and revised at the same time
```
`hosts` can have `meta: {action: 'test'}` without attributes and cannot be modified and revised at the same time.
Updating the examples is on my list of things to do Once purge retries is done.

thebenrob replied:
Added the Hosts example to the bullets.

Commit:
grammatical fixes

Commit:
Added Hosts example to attributes optionsl

thebenrob commented:
So, I think I've addressed all the feedback.  In the diffs, I still see the removal of "children", but it's fixed locally, and local says the branch is up to date.  And if I look at the Files tab, it doesn't show the change there, so I think it's done?  Anyway, please re-review?
